Reneiski_Homework1/ReneiskiSelenium11/Program.cs

```diff
@@ -0,0 +1,246 @@
+            _driverActions.DoubleClick(doubleClickButton).Perform();
+
+            var doubleClickResult = _driverWait.Until(drv => drv.FindElement(By.XPath("//*[@id='doubleClickMessage']")));
+            var doubleClickResultText = doubleClickResult.Text;
+
+            Assert.IsTrue(doubleClickResultText.Contains("You have done a double click"));
+
+            var rightClickButton = _driverWait.Until(drv => drv.FindElement(By.XPath("//*[@id='rightClickBtn']")));
+            _driverActions.ContextClick(rightClickButton).Perform();
+
+            var rightClickResult = _driverWait.Until(drv => drv.FindElement(By.XPath("//*[@id='rightClickMessage']")));
+            var rightClickResultText = rightClickResult.Text;
+
+            Assert.IsTrue(rightClickResultText.Contains("You have done a right click"));
+
+            var dynamicClickButton = _driverWait.Until(drv => drv.FindElement(By.XPath("/html/body/div[2]/div/div/div[2]/div[2]/div[2]/div[3]/button")));
+            dynamicClickButton.Click();
+
+            var dynamicClickResult = _driverWait.Until(drv => drv.FindElement(By.XPath("//*[@id='dynamicClickMessage']")));
+            var dynamicClickResultText = dynamicClickResult.Text;
+
+            Assert.IsTrue(dynamicClickResultText.Contains("You have done a dynamic click"));
+        }
+
+        [Test]
+        public void FifthLinksTest()
+        {
+            _driver.Navigate().GoToUrl("https://demoqa.com/links");
+
+            var simpleLink = _driver.FindElement(By.XPath("//*[@id='simpleLink']"));
+            simpleLink.Click();
+
+            var currentUrl = _driver.FindElement(By.XPath("//p/a[@href='https://demoqa.com']"));
+            var currentUrlResult = currentUrl.GetAttribute("href");
+
+            List<string> windowHandles = _driver.WindowHandles.ToList();
+            _driver.SwitchTo().Window(windowHandles.Last());
+
+            var newUrl = _driver.Url;
+            Assert.AreEqual(newUrl, currentUrlResult);
+
+            _driver.SwitchTo().Window(windowHandles.First());
+
+            var createdLink = _driverWait.Until(drv => drv.FindElement(By.XPath("//*[@id='created']")));
+            createdLink.Click();
+
+            var createdLinkResult = _driverWait.Until(drv => drv.FindElement(By.XPath("//*[@id='linkResponse']")));
+            var createdLinkResultText = createdLinkResult.Text;
+
+            Assert.IsTrue(createdLinkResultText.Contains("Link has responded with staus 201 and status text Created"));
+
+            var noContentLink = _driverWait.Until(drv => drv.FindElement(By.XPath("//*[@id='no-content']")));
+            noContentLink.Click();
+
+            var noContentLinkResult = _driverWait.Until(drv => drv.FindElement(By.XPath("//p[@id='linkResponse']")));
+            var noContentLinkResultText = _driverWait.Until(drv => noContentLinkResult.Text);
+
+            while (noContentLinkResultText != "Link has responded with staus 204 and status text No Content")
```

Comment:
why have you used while + try catch? is there any wait you can add to hangle it without try catch?

Reply:
I added it because wait didn't help (((

Reply:
Not a good solution, please think of how you can change your wait, if no idea - ping me, will have a call to check

Reply:
Fixed